OrbitQt/orbitmainwindow.cpp

```diff
@@ -585,6 +586,26 @@ void OrbitMainWindow::on_actionClear_Capture_triggered() { GOrbitApp->ClearCaptu
 
 void OrbitMainWindow::on_actionHelp_triggered() { GOrbitApp->ToggleDrawHelp(); }
 
+void OrbitMainWindow::on_actionIntrospection_triggered() {
+  if (introspection_widget_ == nullptr) {
+    introspection_widget_ = new OrbitGLWidget();
+    introspection_widget_->setWindowFlags(Qt::WindowStaysOnTopHint);
+    introspection_widget_->Initialize(GlCanvas::CanvasType::kIntrospectionWindow, this, 14);
+    introspection_widget_->installEventFilter(this);
+  }
+
+  introspection_widget_->show();
+}
+
+bool OrbitMainWindow::eventFilter(QObject* object, QEvent* event) {
```

Comment:
Nit: Just FYI, AFAIK the proposed way of doing this would to override the `closeEvent` of OrbitGLWidget and not use the eventFilter. But it's quite annoying that there is no signal emitted on close that you can simply connect to, so I think doing it like this is a workable workaround, but it's somewhat abusing eventFilter IMHO.

Reply:
Ah ok, thanks for the info, I hadn't looked into it in detail. This will change eventually, I was thinking of making a common widget for both the capturewindow and the introspection window so that we could reuse the filters and toolbar.